uxarray/grid/utils.py

```diff
@@ -116,11 +116,22 @@ def _inv_jacobian(x0, x1, y0, y1, z0, z1, x_i_old, y_i_old):
         [ac_utils._fmms(y0, z1, z0, y1), ac_utils._fmms(x0, z1, z0, x1)],
         [2 * x_i_old, 2 * y_i_old],
     ]
+
+    # First check if the Jacobian matrix is singular
+    if np.linalg.matrix_rank(jacobian) < 2:
+        warnings.warn("The Jacobian matrix is singular.")
+        return None
+
     try:
         inverse_jacobian = np.linalg.inv(jacobian)
-    except np.linalg.LinAlgError:
-        raise warnings("Warning: Singular Jacobian matrix encountered.")
-        return None
+    except np.linalg.LinAlgError as e:
+        # Print out the error message
+
+        cond_number = np.linalg.cond(jacobian)
+        print(f"Condition number: {cond_number}")
+        print(f"Jacobian matrix:\n{jacobian}")
+        print(f"An error occurred: {e}")
+        raise
```

Comment:
These look like leftover debugging prints?

Reply:
This is for what happens if an exception occurs. But if you think an user doesn't need this information, you can remove it

Reply:
Sounds good, I can update this (using prints is typically not advised)